lib/Doctrine/DBAL/Schema/AbstractAsset.php

```diff
@@ -48,10 +48,9 @@ protected function _setName($name)
             $this->_quoted = true;
             $name          = $this->trimQuotes($name);
         }
+
         if (strpos($name, '.') !== false) {
-            $parts            = explode('.', $name);
-            $this->_namespace = $parts[0];
-            $name             = $parts[1];
+            [$this->_namespace, $name] = explode('.', $name, 2);
```

Comment:
How do you know that the dot belongs to the asset but not to the namespace? In SQL syntax, a dot is a separator between database name, schema name, object name, etc. It's surprising to see it used as part of the object name, especially in this generic case (any platform, any object type).

Reply:
Hello @morozov,

look at the original version: The part before the first dot was taken for the namespace. The assumption here is, that the rest should be part of the name.